server/api/v1/user/index.patch.ts

```diff
@@ -0,0 +1,34 @@
+import { defineEventHandler, readBody, createError } from "h3";
+import prisma from "~/server/internal/db/database";
+
+export default defineEventHandler(async (event) => {
+  // Get user from session
+  const user = await event.context.session.getUser(event);
+  if (!user) {
+    throw createError({
+      statusCode: 403,
+      statusMessage: "Not authenticated",
+    });
+  }
+
+  const body = await readBody(event);
```

Comment:
We break out body into stuff like this:
```
const body = await readBody(event);
const displayName = body.displayName;
const email = body.email;
```

That lets us do multiple checks without repeating ourselves.